### PR TITLE
youtube-video-duration: filter out videos by their duration

### PR DESCRIPTION
**Pull request by Nomes77**
https://github.com/letsblockit/letsblockit/issues/621

### CODE REVIEW

**Review by JohnyP36 — APPROVED** [review]

**Comment by Nomes77** [comment]
You were right about `^` and `$` not working. However `\b` had some unintentional side effects. 
With `\b` , `\d\d:06` will also block 10:06:28 in stead of only 28:06.
I choose for keeping it a regexp. Hopefully, the explanation is sufficient.

File: data/filters/templates/youtube-video-duration.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+- videos in the subscription and search pages
+- dedicated shelves in the homepage and channel list
+
+The input should be a regexp. In regexp '\d' is for every possible digit. Multiple options should be divided be a '|', some examples:
+
+- less than a minute: 0:\d\d
+- between one and two minutes: 1:\d\d
+- one minute or more: 1:\d\d|\d:\d\d|\d\d:\d\d|\d\d:\d\d:\d\d
+- less than 10 minutes: \d:\d\d
+- 10 minutes or more: 1\d:\d\d|\d:\d\d:\d\d|\d\d:\d\d:\d\d
```
Comment (JohnyP36) [inline data/filters/templates/youtube-video-duration.yaml:50]: What if people want everything between 10 and 25 minutes: `1\d:\d\d|20:\d\d|21:\d\d|22:\d\d|23:\d\d|24:\d\d` would be long. Is there a way to make that shorter?

**Review by JohnyP36 — CHANGES_REQUESTED** [review]

File: data/filters/templates/youtube-video-duration.yaml (hunk 1 above)
Comment (Nomes77) [reply data/filters/templates/youtube-video-duration.yaml:50]: Yes: `1\d:\d\d|(20|21|22|23|24):\d\d`

Comment (Nomes77) [reply data/filters/templates/youtube-video-duration.yaml:50]: or shorter: `1\d:\d\d|2[0-5]:\d\d`

Comment (Nomes77) [reply data/filters/templates/youtube-video-duration.yaml:50]: `1\d:\d\d` will block 10:00-19:59; `\d:\d\d:\d\d` will block 1:00:00-9:59:59; `\d\d:\d\d:\d\d` will block 10:00:00-99:59:59.
`\d` means 0-9.

Comment (JohnyP36) [reply data/filters/templates/youtube-video-duration.yaml:50]: Thus, not between 20:00-59:59 ???

Comment (Nomes77) [reply data/filters/templates/youtube-video-duration.yaml]: Yes, but see https://github.com/letsblockit/letsblockit/pull/622/commits/777e36d266310ec3cd0dafb35563c69c009bbeac

Maybe we should add a link to a website with a good explenation to all regexp syntax rules?

Comment (JohnyP36) [reply data/filters/templates/youtube-video-duration.yaml]: > Maybe we should add a link to a website with a good explenation to all regexp syntax rules?

Probably.

But check line 50
https://github.com/letsblockit/letsblockit/blob/de93b3eca28d4452342b4f3e703c5433870cad72/data/filters/templates/youtube-video-duration.yaml#L50

After the first `\d` is an ) which doesn't have a (. Should it be removed?

Comment (Nomes77) [reply data/filters/templates/youtube-video-duration.yaml]: Yes, done: https://github.com/letsblockit/letsblockit/pull/622/commits/8a7edb6b892643844e403f79e399d221a3326c40

Comment (JohnyP36) [reply data/filters/templates/youtube-video-duration.yaml]: Completely fixed in https://github.com/letsblockit/letsblockit/pull/622/commits/87716edc05499e8df7cf677fe6b52f4dbaf2b767

**Review by JohnyP36 — APPROVED** [review]

File: data/filters/templates/youtube-video-duration.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+title: "YouTube: filter out videos of any duration"
```
Comment (xvello) [inline data/filters/templates/youtube-video-duration.yaml:1]:
```suggestion
title: "YouTube: filter out videos by duration"
```

File: data/filters/templates/youtube-video-duration.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      www.youtube.com##ytd-search ytd-thumbnail #time-status > span:has-text(/\s(0:\d\d)\s/):upward(ytd-video-renderer)
+---
+
+With this filter format it is possible to filter out videos with any duration. This filter hides most
+occurrences of this format:
+
+- videos in the subscription and search pages
+- dedicated shelves in the homepage and channel list
+
+The input should be a regexp. In regexp '\d' is for every possible digit. Multiple options should be divided be a '|', some examples:
```
Comment (xvello) [inline data/filters/templates/youtube-video-duration.yaml:44]:
```suggestion
The input should be a regexp. In regexp '`\d`' is for every possible digit. Multiple options should be separated be a '`|`', some examples:
```

**Review by xvello — APPROVED** [review]
Nice work! Didn't test the rules, but I trust you did.

**Comment by xvello** [comment]
Indeed @Nomes77, `\b` will not work here, as the `:` separator does trigger a word boundary! `\s` it is then.

**Comment by JohnyP36** [comment]
Late reply: yes, the rules work.

**Pull request merged**